### PR TITLE
feat(GES): add a new resource to manage metadata

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccGesMetadata_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ges -v

=== RUN   TestAccGesMetadata_basic
=== PAUSE TestAccGesMetadata_basic
=== CONT  TestAccGesMetadata_basic
--- PASS: TestAccGesMetadata_basic (29.47s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ges	29.510s

```

### CODE REVIEW

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+}
+
+func resourceGesMetadataUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateMetadataChanges := []string{
+		"name",
+		"metadata_path",
+		"is_overwrite",
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: this not contained in the shema

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+			updateMetadataProduct = "ges"
+		)
+		updateMetadataClient, err := cfg.NewServiceClient(updateMetadataProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating GES Client: %s", err)
+		}
+
+		updateMetadataPath := updateMetadataClient.Endpoint + updateMetadataHttpUrl
+		updateMetadataPath = strings.ReplaceAll(updateMetadataPath, "{project_id}", updateMetadataClient.ProjectID)
+		updateMetadataPath = strings.ReplaceAll(updateMetadataPath, "{id}", d.Id())
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: there is no id in updateMetadataHttpUrl

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+	}
+	return rst
+}
+
+func resourceGesMetadataUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateMetadataChanges := []string{
+		"name",
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: looks name should be forcenew and used to distinguish the metadata when updating

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: yes, deleted

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: fixed

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go]: deleted

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+				Description: `Metadata name.`,
+			},
+			"metadata_path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `OBS Path for storing the metadata.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go:57]: is it really required in API?

File: docs/resources/ges_metadata.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+* `labels` - (Optional, List) Label list.  
+  For details, see [data formats](https://support.huaweicloud.com/intl/en-us/usermanual-ges/ges_01_0153.html).
+  The [Labels](#GesMetadata_MetadataLabels) structure is documented below.
+
+<a name="GesMetadata_MetadataLabels"></a>
+The `MetadataLabels` block supports:
+
+* `name` - (Optional, String) Name of a label.
+
+* `properties` - (Optional, List)
```
Comment (ShiChangkuo) [inline docs/resources/ges_metadata.md]: the description is missing

File: huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go:57]: yes

File: docs/resources/ges_metadata.md (hunk 5 above)
Comment (hmbig2) [reply docs/resources/ges_metadata.md]: added

**Pull request merged**